msal/src/main/java/com/microsoft/identity/client/IPublicClientApplication.java

```diff
@@ -84,6 +85,14 @@ void acquireToken(@NonNull final Activity activity,
     @WorkerThread
     IAuthenticationResult acquireTokenSilent(@NonNull final AcquireTokenSilentParameters acquireTokenSilentParameters) throws InterruptedException, MsalException;
 
+    /**
+     * Perform the Device Code Flow (DCF) protocol to allow a device without input capability to authenticate and get a new access token.
+     * Currently, flow is only supported in local MSAL. No Broker support.
+     * @param scopes the desired access scopes
+     * @param callback callback object used to communicate with the API throughout the protocol
+     */
+    void deviceCodeFlow(@Nullable String[] scopes, @NonNull final DeviceCodeFlowCallback callback);
```

Comment:
Should we call this something like [`acquireTokenWithDeviceCode()`](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/1cef43d185bb0a1dcc013d2ec232ae9d4c6c053a/src/client/Microsoft.Identity.Client/IPublicClientApplication.cs#L75)? This seems to be the naming that .NET used

/cc @jennyf19

Reply:
are most of your methods `acquiretokenWith*`? [Java](https://github.com/AzureAD/microsoft-authentication-library-for-java/wiki/Device-Code-Flow#code-snippet) looks to be slightly different too.

Reply:
I think that's a better name for it